examples/surrounding_nitrogen_bonds/nn_bond_length_histogram.py

```diff
@@ -0,0 +1,108 @@
+"""
+Script for generating a histogram based on the bond length of
+single nitrogen-nitrogen bonds
+Usage:
+    python nn_bond_length_histogram.py --smiles_database [DATABASE] --fingerprinted [T or F]
+"""
+
+from openeye import oechem, oeomega, oequacpac, oeff
+import argparse
+import matplotlib.pyplot as plt
+import math
+
+def generateHistogram():
+    """
+    Generates a bond length histogram for single nitrogen-nitrogen bonds
+    using a .smi file
+    """
+    
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--smiles_database",
+                        type=str,
+                        required=True,
+                        help=("Name of a SMILES file containing molecules "
+                              "to generate a bond length histogram."))
+    parser.add_argument("--fingerprinted",
+                        type=str,
+                        required=True,
+                        help=("True or False if the database is fingerprinted or not"))
+    args = parser.parse_args()
+    
+    ifs = oechem.oemolistream(args.smiles_database)
+    subs = oechem.OESubSearch("[#7:1]-[#7:2]")
+    bondLengths = []
+    
+    for molecule in ifs.GetOEMols():
+        mol, status = smiles2oemol(oechem.OEMolToSmiles(molecule))
+        if (status):
+            optimizeMol(mol)
+            
+            match = subs.Match(mol, True)
+            targetBond = match.Target().GetTargetBonds().Target()
+            targetIdxs = (targetBond.GetBgn().GetIdx(), targetBond.GetEnd().GetIdx())
+            
+            n1Coords = mol.GetCoords()[targetIdxs[0]]
+            n2Coords = mol.GetCoords()[targetIdxs[1]]
+            
+            bondLength = getBondLength(n1Coords, n2Coords)
+            bondLengths.append(bondLength)
+    
+    plt.hist(bondLengths, density=True, bins=30)
+    
+    if (args.fingerprinted.lower() == "t"):
+        plt.title(f"Fingerprinted N-N Molecules ({len(bondLengths)} total mols)")
+    elif (args.fingerprinted.lower() == "f"):
+        plt.title(f"Random N-N Molecules ({len(bondLengths)} total mols)")
+    plt.ylabel("Frequency")
+    plt.xlabel("Bond Length (angstrom)")
+    plt.show(block=True)
```

Comment:
how about saving it with `plt.savefig()`?

Reply:
will definitely look into this when I update this script to make more than one plot at once, thank you!